internal/pkg/selfupdate/update.go

```diff
@@ -55,10 +58,20 @@ func UpgradeInPlace(ctx context.Context, curSV semver.Version, baseSiteURL strin
 		return nil
 	} else {
 		if err := cmd.Run(); err != nil {
-			// failed to install new binary, rollback renamed binary's name to previous name
-			os.Rename(renamed, cur)
 			return fmt.Errorf("running update script: %v", err)
 		}
+		currentBinaryPath, err := os.Executable()
+		if err != nil {
+			return fmt.Errorf("getting executable path: %v", err)
+		}
+		newBinaryPath := filepath.Join(tmp, "bin", "humanlog") // command 'install.sh | sh' makes new directory 'bin' and install binary under it
+		_, err = os.Stat(newBinaryPath)
+		if err != nil {
+			return fmt.Errorf("can't find installed binary: %v", err)
```

Comment:
`err` here can about many things other than `can't find`. The error string should be less specific, or you should check that the `errors.Is(os.ErrNotExists, err)` and return a specific error in that case.

Reply:
got it